packages/toolkit/src/view/knowledge/components/EditKnowledgeDialog.tsx

```diff
@@ -36,24 +46,26 @@ export const EditKnowledgeDialog = ({
 }: EditKnowledgeDialogProps) => {
   const [isSubmitting, setIsSubmitting] = React.useState(false);
 
-  const form = useForm<z.infer<typeof EditKnowledgeFormSchema>>({
+  const form = useForm<EditKnowledgeDialogData>({
     resolver: zodResolver(EditKnowledgeFormSchema),
     defaultValues: initialValues,
     mode: "onChange",
   });
 
-  const { formState, reset } = form;
+  const { formState, reset, watch } = form;
+  const description = watch("description");
 
-  const handleSubmit = async (
-    data: z.infer<typeof EditKnowledgeFormSchema>,
-  ) => {
+  React.useEffect(() => {
+    if (isOpen) {
+      reset(initialValues);
+    }
+  }, [isOpen, initialValues, reset]);
+
+  const handleSubmit = async (data: EditKnowledgeDialogData) => {
     setIsSubmitting(true);
     try {
       await onSubmit(data);
-      setTimeout(() => {
-        onClose();
-        reset(initialValues);
-      }, 1000);
+      onClose();
     } catch (error) {
       console.error("Error updating catalog:", error);
     } finally {
```

Comment:
@thewbuk You don't need finally here. Just put the setState after or before onClose